File: spider/spider-jar/src/main/java/com/github/catvod/crawler/Spider.java
package com.github.catvod.crawler;

import android.content.Context;
import java.io.InputStream;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public abstract class Spider {
    public void init(Context context, String extend) throws Exception {}
    public String homeContent(boolean filter) throws Exception { return ""; }
    public String homeVideoContent() throws Exception { return ""; }
    public String categoryContent(String tid, String pg, boolean filter, HashMap<String, String> extend) throws Exception { return ""; }
    public String detailContent(List<String> ids) throws Exception { return ""; }
    public String searchContent(String key, boolean quick) throws Exception { return ""; }
    public String searchContent(String key, boolean quick, String pg) throws Exception { return ""; }
    public String playerContent(String flag, String id, List<String> vipFlags) throws Exception { return ""; }
    public boolean manualVideoCheck() throws Exception { return false; }
    public boolean isVideoFormat(String url) throws Exception { return false; }
    public Object[] proxyLocal(HashMap<String, String> params) throws Exception { return new Object[]{}; }
    public Object[] proxyLocal(Map<String, String> params) throws Exception { return new Object[]{}; }
    public Object[] proxy(HashMap<String, String> params) throws Exception { return proxyLocal(params); }
    public Object[] proxy(Map<String, String> params) throws Exception { return proxyLocal(params); }
    public void cancelByTag(String tag) {}
    public void destroy() {}
}

File: spider/spider-api/src/main/kotlin/com/github/tvbox/newbox/spider/api/Spider.kt
package com.github.tvbox.newbox.spider.api

import android.content.Context

interface Spider {
    suspend fun init(context: Context, extend: String) {}
    suspend fun homeContent(filter: Boolean): String
    suspend fun homeVideoContent(): String = ""
    suspend fun categoryContent(tid: String, pg: String, filter: Boolean, extend: Map<String, String>): String
    suspend fun detailContent(ids: List<String>): String
    suspend fun searchContent(key: String, quick: Boolean, pg: String = "1"): String
    suspend fun playerContent(flag: String, id: String, vipFlags: List<String>): String
    suspend fun proxyLocal(params: Map<String, String>): String = ""
    suspend fun proxy(params: Map<String, String>): String = ""
    fun cancelByTag(tag: String) {}
    fun destroy() {}
}
